### PR TITLE
[v2] migrate to Operator SDK v1.0.0

### LINKED ISSUE #969: Migrate to the new Kubebuilder based Operator SDK framework
From v0.19 Operator SDK project was merged with Kubebuilder, this results in a new project layout (ie. KEDA code layout at build stuff).

We should migrate to the new project structure for v2.

https://sdk.operatorframework.io/docs/golang/quickstart/

### PR DESCRIPTION
**Pull request by zroubalik**
Signed-off-by: Zbynek Roubalik <zroubali@redhat.com>

OMG, sorry for the huge PR, there wasn't an easy way how to split things.

Operator SDK is finally in 1.0.0 version, this brought change in the project layout and other stuff. Now it is internally using [Kubebuilder](https://book.kubebuilder.io/) for project scaffolding and [kustomize](https://kubernetes-sigs.github.io/kustomize/) to generate YAML manifests. See the migration guide for details:
https://sdk.operatorframework.io/docs/building-operators/golang/project_migration_guide/

### What was changed

- The `deploy` directory was replaced with the `config` directory including a new layout of Kubernetes manifests files:
  - CRD manifests in `deploy/crds/` are now in `config/crd/bases`
  - CR manifests in `deploy/crds/` are now in `config/samples`
  - Controller manifest `deploy/operator.yaml` is now in `config/manager/manager.yaml`
  - RBAC manifests in `deploy` are now in `config/rbac/`
  - Metrics server manifests are now in `config/metrics-server`
- `build/Dockerfile` and`build/Dockerfile.adapter` are moved to the project root directory
  - Dockerfiles were changed to multi-stage and distroless, this should reduce image size
- `pkg/apis` and `pkg/controllers` are now in the root directory.
  - there were only minor changes in the api and controllers code
- `cmd/manager/main.go` is now in the root director
- `cmd/adapter/main.go` is now in `adapter/main.go`
- [kustomize](https://kubernetes-sigs.github.io/kustomize/) is used to manage Kubernetes resources needed to deploy your operator
  - manifests are generated with proper version/image tags as specified in `Makefile`
  - ScaledJob [generation](https://github.com/kedacore/keda/issues/927) is fixed for now (Kustomize enables us to use patches: https://github.com/kedacore/keda/blob/950c5e16deb143a8201189ec7430f7350d0fa6bd/config/crd/patches/scaledjob_patch.yaml)
- `Makefile` was modifed a lot, eg. you can use `make deploy` to deploy KEDA, `make run` to run locally and other useful steps
- Release process was simplified (no need to manually change version etc.) everything should be part of GitHub Release Action now
- We produce one release yaml file that contains all resources (eg. `keda-2.0.0.yaml`) instead of `.zip` or `tar.gz` artifacts
- We could use [envtest test framework](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/envtest) to test our controllers. The current test file is just a stub: https://github.com/kedacore/keda/blob/950c5e16deb143a8201189ec7430f7350d0fa6bd/controllers/suite_test.go
  - issue created: https://github.com/kedacore/keda/issues/1008

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)
- [x] Changelog has been updated

Fixes #969

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -50,95 +59,179 @@ e2e-test:
...
+controller-gen:
+ifeq (, $(shell which controller-gen))
+	@{ \
+	set -e ;\
+	CONTROLLER_GEN_TMP_DIR=$$(mktemp -d) ;\
+	cd $$CONTROLLER_GEN_TMP_DIR ;\
+	go mod init tmp ;\
+	go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.3.0 ;\
+	rm -rf $$CONTROLLER_GEN_TMP_DIR ;\
+	}
```
Comment (ahmelsayed) [inline Makefile:170]: just curious if 

```
GO111MODULE=on go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.3.0
```

could be used here, or if this tmp directory is needed.

File: go.mod [hunk 2]
```diff
@@ -61,3 +60,6 @@ replace (
 	google.golang.org/genproto => google.golang.org/genproto v0.0.0-20191002211648-c459b9ce5143
 	google.golang.org/grpc => google.golang.org/grpc v1.24.0
 )
+
+// WORKAROUND until k8s v1.18+ is not present in knative/pkg
```
Comment (ahmelsayed) [inline go.mod:64]: Just curious to understand this a bit more. do you mean is present, or is there a particular issue with 1.18?

Comment (zroubalik) [reply go.mod:64]: haha, sorry for the confusing wording :) I mean is present (the client api was changed a little bit in k8s 1.18), knative should move to k8s in a near future. I am gonna rephrase the comment

File: Makefile (hunk 1 above)
Comment (zroubalik) [reply Makefile]: Good catch, you are right. This part of the Makefile is directly from the Kubebuilder scaffolding. I have changed it as you suggested, if there's any problem in the future, we can revert back.

**Pull request merged**